lib/tasks/instrumentation_generator/templates/dependency_detection.tt

```diff
@@ -18,6 +14,10 @@ DependencyDetection.defer do
   end
 
   executes do
+    require_relative '<%= @snake_name.downcase %>/instrumentation'
+    require_relative '<%= @snake_name.downcase %>/chain'
+    require_relative '<%= @snake_name.downcase %>/prepend'
+
```

Comment:
Thoughts on going a step further... to require the instrumentation file here, and the chain and/or prepend files within the condition below? 

```
require_relative '<%= @snake_name.downcase %>/instrumentation'

  if use_prepend? 
    require_relative '<%= @snake_name.downcase %>/prepend'
    prepend_instrument <%= @class_name %>, NewRelic::Agent::Instrumentation::<%= @class_name %>::Prepend
 else
   require_relative '<%= @snake_name.downcase %>/chain'
   chain_instrument NewRelic::Agent::Instrumentation::<%= @class_name %>::Chain
 end
 ```

Reply:
Sure, addressed with [1c6d8a3](https://github.com/newrelic/newrelic-ruby-agent/pull/2917/commits/1c6d8a35f40240168f043379b4d9f9947fa89877).